fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/TriState.java

```diff
@@ -16,16 +16,197 @@
+	 *
+	 * @return {@code null} if {@link TriState#DEFAULT}.
+	 * Otherwise {@code true} if {@link TriState#TRUE} or {@code false} if {@link TriState#FALSE}.
+	 */
+	/* @Nullable */
+	public Boolean getBoxed() {
+		if (this == TriState.DEFAULT) {
+			return null;
+		}
+
+		return this.get();
+	}
+
+	/**
+	 * Gets the value of this tri-state.
+	 * If the value is {@link TriState#DEFAULT} then use the supplied value.
+	 *
+	 * @param value the value to fallback to
+	 * @return the value of the tri-state or the supplied value if {@link TriState#DEFAULT}.
+	 */
+	public boolean orElse(boolean value) {
+		return this == TriState.DEFAULT ? value : this.get();
+	}
+
+	/**
+	 * Gets the value of this tri-state.
+	 * If the value is {@link TriState#DEFAULT} then use the supplied value.
+	 *
+	 * @param supplier the supplier used to get the value to fallback to
+	 * @return the value of the tri-state or the value of the supplier if the tri-state is {@link TriState#DEFAULT}.
+	 */
+	public boolean orElseGet(BooleanSupplier supplier) {
+		return this == TriState.DEFAULT ? supplier.getAsBoolean() : this.get();
+	}
+
+	/**
+	 * Maps the value of an optional if this tri-state is {@link TriState#TRUE}.
+	 *
+	 * @param mapper the mapper to use
+	 * @param <T> the type of object supplied by the mapper
+	 * @return an optional containing the value if {@link TriState#TRUE} and the value returned by the mapper is non-null, otherwise {@link Optional#empty()}.
+	 */
+	public <T> Optional<T> mapTrue(Supplier<T> mapper) {
+		Objects.requireNonNull(mapper);
+
+		if (this == TRUE) {
+			return Optional.ofNullable(mapper.get());
+		}
+
+		return Optional.empty();
+	}
+
+	/**
+	 * Maps the value of an optional if this tri-state is {@link TriState#TRUE}.
+	 *
+	 * @param mapper the mapper to use
+	 * @param <T> the type of object supplied by the mapper
+	 * @return an optional containing the value if {@link TriState#FALSE} and the value returned by the mapper is non-null, otherwise {@link Optional#empty()}.
+	 */
+	public <T> Optional<T> mapFalse(Supplier<T> mapper) {
+		Objects.requireNonNull(mapper);
+
+		if (this == FALSE) {
+			return Optional.ofNullable(mapper.get());
+		}
+
+		return Optional.empty();
+	}
+
+	/**
+	 * Maps the value of an optional if this tri-state is {@link TriState#DEFAULT}.
+	 *
+	 * @param mapper the mapper to use
+	 * @param <T> the type of object supplied by the mapper
+	 * @return an optional containing the value if {@link TriState#DEFAULT} and the value returned by the mapper is non-null, otherwise {@link Optional#empty()}.
+	 */
+	public <T> Optional<T> mapDefault(Supplier<T> mapper) {
+		Objects.requireNonNull(mapper);
+
+		if (this == DEFAULT) {
+			return Optional.ofNullable(mapper.get());
+		}
+
+		return Optional.empty();
+	}
+
+	/**
+	 * Calls the supplied runnable if this tri-state is {@link TriState#TRUE}.
+	 *
+	 * @param runnable the interface to run
+	 */
+	public void ifTrue(Runnable runnable) {
+		if (this == TRUE) {
```

Comment:
Same here you could probably just make one general method and have the three enum type corresponding methods just use that one general method's return with the respective enum type as a parameter.

```java
private void ifEq(TriState comparison, Runnable runnable) {
    if (this == comparison) {
        runnable.run();
    }
}

public void ifTrue(Runnable runnable) {
    this.ifEq(TriState.TRUE, runnable);
}

public void ifFalse(Runnable runnable) {
    this.ifEq(TriState.FALSE, runnable);
}

public void ifDefault(Runnable runnable) {
    this.ifEq(TriState.DEFAULT, runnable);
}
```